RecoTracker/MkFitCore/interface/Track.h

```diff
@@ -0,0 +1,800 @@
+#ifndef RecoTracker_MkFitCore_interface_Track_h
+#define RecoTracker_MkFitCore_interface_Track_h
+
+#include "RecoTracker/MkFitCore/interface/Config.h"
+#include "RecoTracker/MkFitCore/interface/MatrixSTypes.h"
+#include "RecoTracker/MkFitCore/interface/Hit.h"
+#include "RecoTracker/MkFitCore/interface/TrackerInfo.h"
+
+#include <vector>
+#include <map>
+
+namespace mkfit {
+
+  typedef std::pair<int, int> SimTkIDInfo;
+  typedef std::vector<int> HitIdxVec;
+  typedef std::map<int, std::vector<int> > HitLayerMap;
+
+  inline int calculateCharge(const Hit& hit0, const Hit& hit1, const Hit& hit2) {
+    return ((hit2.y() - hit0.y()) * (hit2.x() - hit1.x()) > (hit2.y() - hit1.y()) * (hit2.x() - hit0.x()) ? 1 : -1);
+  }
+
+  inline int calculateCharge(const float hit0_x,
+                             const float hit0_y,
+                             const float hit1_x,
+                             const float hit1_y,
+                             const float hit2_x,
+                             const float hit2_y) {
+    return ((hit2_y - hit0_y) * (hit2_x - hit1_x) > (hit2_y - hit1_y) * (hit2_x - hit0_x) ? 1 : -1);
+  }
+
+  struct IdxChi2List {
+  public:
+    int trkIdx;           // candidate index
+    int hitIdx;           // hit index
+    unsigned int module;  // module id
+    int nhits;            // number of hits (used for sorting)
+    int ntailholes;       // number of holes at the end of the track (used for sorting)
+    int noverlaps;        // number of overlaps (used for sorting)
+    int nholes;           // number of holes (used for sorting)
+    float pt;             // pt (used for sorting)
+    float chi2;           // total chi2 (used for sorting)
+    float chi2_hit;       // chi2 of the added hit
+    float score;          // score used for candidate ranking
+  };
+
+  //==============================================================================
+  // ReducedTrack
+  //==============================================================================
+
+  struct ReducedTrack  // used for cmssw reco track validation
+  {
+  public:
+    ReducedTrack() {}
+    ReducedTrack(const int label, const int seedID, const SVector2& params, const float phi, const HitLayerMap& hitmap)
+        : label_(label), seedID_(seedID), parameters_(params), phi_(phi), hitLayerMap_(hitmap) {}
+    ~ReducedTrack() {}
```

Comment:
use default?

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit https://github.com/trackreco/cmssw/pull/76/commits/a52b1a9d88304f2becee4962fa1f431c24020fdc